src/scripts/main.js

```diff
@@ -1,5 +1,29 @@
 'use strict';
 
+const wall = document.querySelector('.wall');
+const spider = document.querySelector('.spider');
+
 document.addEventListener('click', e => {
-  // write code here
+  const wallXCorrection = wall.getBoundingClientRect().left;
+  const wallYCorrection = wall.getBoundingClientRect().top;
+  let xPos = e.clientX - (spider.clientWidth / 2)
+  - wallXCorrection;
+  let yPos = e.clientY - (spider.clientHeight / 2)
+  - wallYCorrection;
+
+  if (xPos < 0) {
+    xPos = 0;
+  } else if (xPos > wall.clientWidth - spider.clientWidth) {
```

Comment:
create a variable for `wall.clientWidth - spider.clientWidth`, because this piece of code repeats several times

Reply:
Good points, thanks.